### DISCUSSION TITLE
Static definition of annotators in scripts/global_state.py

### DISCUSSION THREAD

**Discussion by TheOnlyHolyMoly** [Mikubill/sd-webui-controlnet#996] [Q&A] [answered]
Hello, this is my first post in Github. 

I have tried to clear up controlnet files with the release of controlnet 1.1 on my local installation to avoid building up of available annotators. However I was left with a number of controlnet annotators in the UI selector that I could not get rid of. Thought it was a mistake so I completely reinstalled controllnet nightly and there they were again. 

I have tracked it down to the static list in scripts/global_state.py which is also the same as in the current repo and I was just wondering, what the sense of the static definition is?

**Comment by ljleb** [comment, marked as answer]
We currently use `global_state.cn_preprocessor_modules` to determine the available entries in the preprocessor dropdown of the ui.

At the moment, removing or adding preprocessors is not officially supported. We could eventually add something in the ui or programmaticaly to manage the preprocessors. For now, if you just want to remove a preprocessor from the ui, you can manually remove entries from the `global_state.cn_preprocessor_modules` dict. You'll also have to edit `preprocessor_aliases` to prevent the corresponding alias name from showing up.

**Reply by TheOnlyHolyMoly** [reply]
Thanks! I did not want to loose them so I turned them into annotations for the time being.
